ee/clickhouse/models/person.py

```diff
@@ -113,6 +113,35 @@ def delete_person(
     sync_execute(DELETE_PERSON_BY_ID, data)
 
 
+# counts duplicate distinct IDs in clickhouse for a team broken down by day
+def count_duplicate_distinct_ids_for_team(team_id: Union[str, int]) -> Dict:
+    query_result = sync_execute(
+        """
+        SELECT count(1) as ids_with_duplicates, minus(sum(count), ids_with_duplicates) as extra_rows
+        FROM (
+            SELECT distinct_id, count(*) as count
+            FROM (
+                SELECT person_id, distinct_id, max(_timestamp) as timestamp
+                FROM person_distinct_id
+                WHERE team_id = %(team_id)s
+                GROUP BY person_id, distinct_id, team_id
+                HAVING max(is_deleted) = 0
+            )
+            GROUP BY distinct_id
+            HAVING count > 1
+        )
+        """,
+        {"team_id": str(team_id)},
+    )
+    # total_distinct_ids_with_duplicates = how many distinct IDs point to more than one person
+    # total_extra_distinct_id_rows = number of undeleted distinct ID rows that we shouldn't have
+    result = {
+        "total_distinct_ids_with_duplicates": query_result[0][0],
```

Comment:
The case of having an anonymous user (with anon ID) and then having it identified seems quite common, maybe worth also reporting for users who have more than 2 distinct IDs? Think that 2 distinct IDs for a single person is quite acceptable.

Reply:
Oh, this isn't for a single person, sorry if the name is confusing.

This groups by `person_id`, so this number only shows us distinct IDs pointing to multiple different people

Reply:
Yeah no, that's clear. I meant that a user having more than one distinct ID could still be quite legitimate. Particularly if they have two. So wondering if it was worth also reporting number of persons with more than two distinct IDs

Reply:
Ah right, so that's a separate issue - but yes, could add reporting for that too

Reply:
Honestly you have a lot more context on this general issue than me, so if you think this brings enough info to better size the problem and debug, please go for it! From my very limited perspective, I just thought there could be legitimate cases for persons have multiple IDs.

Reply:
No but I think there's some confusion here.

There are legitimate cases for people having multiple IDs, but that's excluded here.

Here we only look at the number of distinct IDs that point to more than one distinct person ID. That should never happen

Reply:
Oh okay, really sorry I misunderstood what we were capturing here, yes, perfect lgtm!